src/main/java/com/empowerops/linqalike/queries/WithQuery.java

```diff
@@ -0,0 +1,69 @@
+package com.empowerops.linqalike.queries;
+
+import com.empowerops.linqalike.DefaultedQueryable;
+import com.empowerops.linqalike.common.EqualityComparer;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static com.empowerops.linqalike.Factories.from;
+import static com.empowerops.linqalike.queries.Accessors.vSize;
+
+/**
+ * Created by Geoff on 2015-12-03.
+ */
+public class WithQuery<TElement> implements DefaultedQueryable<TElement>, FastSize {
+
+    private final Iterable<? extends TElement> left;
+    private final Iterable<? extends TElement> right;
+
+    public WithQuery(Iterable<? extends TElement> left, TElement[] right) {
+        this.left = left;
+        this.right = from(right);
+    }
+
+    public WithQuery(Iterable<? extends TElement> left, Iterable<? extends TElement> right) {
+        this.left = left;
+        this.right = right;
+    }
+
+    @Override
+    public Iterator<TElement> iterator() {
+        return new WithQueryIterator();
```

Comment:
I think I'm formally retiring the nested class iterator in favor of the inner class, its more convenient -> more readable, and I don't know how you would manage to serialize an iterator --and even if you did, any serializer worth its salt is going to capture the enclosing class.